pkg/contexts/ocm/blobhandler/handlers/generic/mvn/blobhandler.go

```diff
@@ -0,0 +1,190 @@
+package mvn
+
+import (
+	"context"
+	"crypto"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/mandelsoft/vfs/pkg/vfs"
+
+	"github.com/open-component-model/ocm/pkg/contexts/credentials/builtin/mvn/identity"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/accessmethods/mvn"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/cpi"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/resourcetypes"
+	"github.com/open-component-model/ocm/pkg/logging"
+	"github.com/open-component-model/ocm/pkg/mime"
+	"github.com/open-component-model/ocm/pkg/utils/tarutils"
+)
+
+const BLOB_HANDLER_NAME = "ocm/" + resourcetypes.MVN_ARTIFACT
+
+type artifactHandler struct {
+	spec *Config
+}
+
+func NewArtifactHandler(repospec *Config) cpi.BlobHandler {
+	return &artifactHandler{repospec}
+}
+
+var log = logging.Context().Logger(identity.REALM)
```

Comment:
If you use a persistent logger, you should create a dynamic logger, because otherwise the logging settings are only evaluate at the time the logger is created.

Reply:
follow up required

Reply:
done